pkg/adhoc/server/server.go

```diff
@@ -0,0 +1,119 @@
+package server
+
+import (
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/fs"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+
+	"github.com/pyroscope-io/pyroscope/pkg/adhoc/util"
+)
+
+type profile struct {
+	ID        string    `json:"id"`
+	Name      string    `json:"name"`
+	UpdatedAt time.Time `json:"updated_at"`
+}
+
+type server struct {
+	log      logrus.FieldLogger
+	enabled  bool
+	profiles map[string]profile
+	mtx      sync.RWMutex
+}
+
+type Server interface {
+	AddRoutes(router *mux.Router) http.HandlerFunc
+}
+
+func New(log logrus.FieldLogger, enabled bool) Server {
+	return &server{log: log, enabled: enabled, profiles: make(map[string]profile)}
+}
+
+func (s *server) AddRoutes(r *mux.Router) http.HandlerFunc {
+	if s.enabled {
+		r.HandleFunc("/v1/profiles", s.Profiles)
+		r.HandleFunc("/v1/profile/{id:[0-9a-f]+}", s.Profile)
+		r.HandleFunc("/v1/diff/{id1[0-9a-f]+}/{id2[0-9a-f]+}", s.Diff)
```

Comment:
I would love to start prefixing all of these with `/api`, I think you can do that globally with gorilla.

One immediate case is https://github.com/pyroscope-io/pyroscope/blob/main/scripts/webpack/webpack.dev.js#L31

Reply:
Done at https://github.com/pyroscope-io/pyroscope/pull/624/commits/f1d205dbbdf35945f8acdd803d1c7529ac863992